server/src/main/java/io/druid/segment/loading/LocalDataSegmentPusher.java

```diff
@@ -54,7 +54,7 @@ public LocalDataSegmentPusher(
   @Override
   public String getPathForHadoop(String dataSource)
   {
-    return String.format("file://%s/%s", config.getStorageDirectory(), dataSource);
+    return String.format("file://%s/%s", config.getStorageDirectory().getAbsoluteFile(), dataSource);
```

Comment:
I was avoiding the Hadoop classes since this isn't a hadoop related module. Could probably use the Java stuff a bit more though, let me see